zio-http/src/main/scala/zio/http/Body.scala

```diff
@@ -170,19 +209,27 @@ object Body {
   /**
    * Helper to create Body from Stream of bytes
    */
-  def fromStream(stream: ZStream[Any, Throwable, Byte]): Body = {
-    new Body {
-      override def isComplete: Boolean = false
+  def fromStream(stream: ZStream[Any, Throwable, Byte]): Body = new StreamBody(stream)
 
-      override def asChunk(implicit trace: Trace): Task[Chunk[Byte]] = stream.runCollect
+  private[zio] final case class StreamBody(stream: ZStream[Any, Throwable, Byte]) extends AnyVal with Body {
 
-      override def asStream(implicit trace: Trace): ZStream[Any, Throwable, Byte] = stream
+    override def asArray(implicit trace: Trace): Task[Array[Byte]] = asChunk.map(_.toArray)
 
-      override def write(ctx: Ctx)(implicit trace: Trace): Task[Boolean] =
-        for {
-          _ <- stream.runForeachChunk(c => ZIO.succeed(ctx.writeAndFlush(Unpooled.wrappedBuffer(c.toArray))))
-          _ <- ZIO.attempt(ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT))
-        } yield true
+    override def isComplete: Boolean = false
+
+    override def asChunk(implicit trace: Trace): Task[Chunk[Byte]] = stream.runCollect
+
+    override def asStream(implicit trace: Trace): ZStream[Any, Throwable, Byte] = stream
+
+    override private[zio] def asChunkUnsafe(implicit unsafe: Unsafe): Chunk[Byte] = {
+      implicit val trace = Trace.empty
+      val rtm            = Runtime.default
+      val exit           = rtm.unsafe.run(asChunk)
```

Comment:
This unsafe run is going to be quite slow. I feel like rather than doing an unsafe run here, we should do it closer to Netty, where there exists the possibility we can collapse many unsafe run into a single one.